### PR TITLE
Updates to metrics_impl

### PR DESCRIPTION
**Pull request by n2cholas**
Description:

Addresses comments in #1238 and prevents metrics from being accumulated on an XLA device.

Check list:
* [x] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/accumulation.py [hunk 1]
```diff
@@ -51,6 +51,8 @@ def __init__(
         self.num_examples = None
         self._op = op
 
+        if torch.device(device).type == "xla":
```
Comment (vfdev-5) [inline ignite/metrics/accumulation.py:54]: @n2cholas Maybe we can do this check once inside `Metric.__init__`, what do you think ?

**Review by vfdev-5 — COMMENTED** [review]
@n2cholas thanks for the updating PR ! I left a comment and fixed isort issue.

File: ignite/metrics/accumulation.py (hunk 1 above)
Comment (n2cholas) [reply ignite/metrics/accumulation.py:54]: I initially had this line in `Metric.__init__`, but noticed that the `ConfusionMatrix` had tests on XLA, so I did it this way instead so I didn't break those tests. I would definitely prefer moving that line to `Metric.__init__`, if it's ok to break the existing `ConfusionMatrix` functionality. What do you think?

Comment (vfdev-5) [reply ignite/metrics/accumulation.py:54]: Sorry for delayed answer. 

> if it's ok to break the existing ConfusionMatrix functionality

I think we can do that. You are talking about this test, right ? 
https://github.com/pytorch/ignite/blob/d192a8f43c63e7da922110474603389b1c0e030c/tests/ignite/metrics/test_confusion_matrix.py#L617

Comment (n2cholas) [reply ignite/metrics/accumulation.py:54]: Yes, I was talking about that test. Sounds good, I'll work on moving the check to `Metric.__init__`!

File: ignite/metrics/precision.py [hunk 2]
```diff
@@ -127,18 +127,6 @@ def thresholded_output_transform(output):
             default, CPU.
 
     """
-
-    def __init__(
```
Comment (vfdev-5) [inline ignite/metrics/precision.py:131]: I think this was kept for proper docs rendering. Can you please check that ?

**Review by vfdev-5 — COMMENTED** [review]
Looks good ! Thanks @n2cholas  !
Remains to fix black formatting and the question about Precision/Recall `__init__`.

File: ignite/metrics/precision.py (hunk 2 above)
Comment (n2cholas) [reply ignite/metrics/precision.py]: Reverted the change! The Precision and Recall docstrings render like before now.

**Comment by n2cholas** [comment]
@vfdev-5 Done! Everything's black formatted and the Precision/Recall docstrings look like they did before.

**Comment by vfdev-5** [comment]
@n2cholas thanks ! Let CI do its job and we can merge it to the branch for GPU checkings...
I think we are almost done with the FR, right ?

**Comment by n2cholas** [comment]
@vfdev-5 Yes, I believe so! We discussed adding tests for #1242, but I think that should be a separate PR (since there's discussion on that issue about ways to fix the problem more generally).

**Comment by vfdev-5** [comment]
I agree. Maybe, it would be good to add a test for `MetricLambda` if components has different devices, https://github.com/pytorch/ignite/pull/1238#discussion_r485717121

**Pull request merged**